public/pages/PolicyDetails/containers/PolicyDetails/PolicyDetails.tsx

```diff
@@ -0,0 +1,243 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+import React, { Component } from "react";
+import {
+  EuiSpacer,
+  EuiTitle,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButton,
+  EuiOverlayMask,
+  EuiButtonEmpty,
+  EuiModalFooter,
+  EuiModal,
+  EuiModalHeader,
+  EuiModalHeaderTitle,
+  EuiModalBody,
+  EuiCodeBlock,
+} from "@elastic/eui";
+import { RouteComponentProps } from "react-router-dom";
+import queryString from "query-string";
+import { PolicyService } from "../../../../services";
+import { BREADCRUMBS, ROUTES } from "../../../../utils/constants";
+import { getErrorMessage } from "../../../../utils/helpers";
+import PolicySettings from "../../components/PolicySettings/PolicySettings";
+import { ISMTemplate } from "../../../../../models/interfaces";
+import { CoreServicesContext } from "../../../../components/core_services";
+import DeleteModal from "../../components/DeleteModal/DeleteModal";
+import States from "../../../VisualCreatePolicy/components/States";
+
+interface PolicyDetailsProps extends RouteComponentProps {
+  policyService: PolicyService;
+}
+
+interface PolicyDetailsState {
+  policyId: string;
+  channelId: string;
+  primaryTerm: number;
+  lastUpdated: string;
+  description: string;
+  sequenceNumber: number;
+  schemaVersion: number;
+  ismTemplates: ISMTemplate[];
+
+  isJSONModalOpen: boolean;
+  policyJSON: any;
+  isDeleteModalVisible: boolean;
+}
+
+export default class PolicyDetails extends Component<PolicyDetailsProps,
+  PolicyDetailsState> {
+    static contextType = CoreServicesContext;
+    constructor(props: PolicyDetailsProps) {
+      super(props);
+
+      this.state = {
+        policyId: "",
+        channelId: "",
+        primaryTerm: -1,
+        lastUpdated: "",
+        description: "",
+        sequenceNumber: -1,
+        schemaVersion: -1,
+        ismTemplates: [],
+
+        isJSONModalOpen: false,
+        policyJSON: {policy:{states:[]}},
```

Comment:
Should default this to null

Reply:
If I default this to null, I'm not sure how to handle that in the rest of the page. The page is trying to render before the API returns, at which point it encounters the null and throws exceptions before it's able to get the actual policy and render correctly. Not sure how to handle that, or how to get the page to wait until that API returns.